src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,164 @@
+import {
+  FC, useCallback, useEffect, useMemo, useState,
+} from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { getPeople } from '../api';
+import { Person } from '../types/Person';
+
+function centuryFromYear(year: number): string {
+  return `${Math.floor((year - 1) / 100) + 1}`;
+}
+
+export const PeoplePage: FC = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoad, setIsLoad] = useState(false);
+  const [errorLoad, setErroreLoad] = useState(false);
+  const [isActiveSort, setIsActiveSort] = useState(false);
+
+  const query = searchParams.get('query') || null;
+  const sexFilter = searchParams.get('sexFilter') || null;
+  const centuries = searchParams.getAll('centuries') || [];
+
+  const order = searchParams.get('order') || null;
+  const sort = searchParams.get('sort') || null;
+
+  // console.log(order, 'on page');
+
+  useEffect(() => {
+    setIsLoad(true);
+    getPeople()
+      .then((res) => {
+        setPeople(res);
+      })
+      .catch(() => setErroreLoad(true))
+      .finally(() => setIsLoad(false));
+  }, []);
+
+  const activateSort = (param: boolean) => setIsActiveSort(param);
+
+  const filtredPeople = useCallback((peopleToFilter: Person[]): Person[] => {
+    let copy = [...peopleToFilter];
```

Comment:
Why are you using the useCallback hook and then useMemo for the sort function? it's better to useMemo right away.

Reply:
+

Reply:
plus

Reply:
or move this function to separate file